downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/Biopax.java

```diff
@@ -0,0 +1,173 @@
+package org.reactome.release.downloadDirectory;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.biopax.validator.api.Validator;
+import org.biopax.validator.api.ValidatorUtils;
+import org.biopax.validator.api.beans.Validation;
+import org.biopax.validator.impl.IdentifierImpl;
+import org.jdom.*;
+import org.reactome.biopax.SpeciesAllPathwaysConverter;
+import org.reactome.biopax.SpeciesAllPathwaysLevel3Converter;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+import org.springframework.core.io.Resource;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+
+public class Biopax {
+  private static final Logger logger = LogManager.getLogger();
+  static ApplicationContext ctx;
+  static boolean autofix = false;
+  static int maxErrors = 0;
+  static String profile = "notstrict";
+  static String outFormat = "xml";
+  static Validator validator;
+
+  static {
+   logger.info("Running BioPAX");
+    // Biopax validation requires loading of bio ontologies, which takes a few minutes
+    ctx = new ClassPathXmlApplicationContext(new String[]{
+      "META-INF/spring/appContext-validator.xml",
+      "META-INF/spring/appContext-loadTimeWeaving.xml"
+    });
+    validator = (Validator) ctx.getBean("validator");
+  }
+  
+  public static void execute(String username, String password, String host, String port, String database, String releaseNumber) throws Exception {
+	  	releaseNumber = releaseNumber.replace("/", "");
+	  	String biopaxDir = releaseNumber + "_biopax";
+        File biopaxDirFile = new File(biopaxDir);
+        if (!biopaxDirFile.exists()) {
+        	biopaxDirFile.mkdir();
+        }
+        // First, generate owl files. This particular step requires a local maven installation of the PathwayExchange jar
+		for (int i = 0; i < 2; i++) {
```

Comment:
The 'i' variable could be replaced with 'bioPaxLevel' directly:

`for (int biopaxLevel = 2; biopaxLevel <= 3; biopaxLevel++)`

Reply:
Yea thats a good point, I originally thought to do `i=2, i < 3; i++`, but thought it would be confusing. I'll make that change as well

Reply:
Or a function call that takes the biopax level as an argument...

Reply:
That sounds better.  That way the loop could be replaced by two function calls.